components/eamxx/src/share/io/scream_scorpio_interface.cpp

```diff
@@ -338,11 +338,20 @@ void register_variable(const std::string &filename, const std::string& shortname
           " - pio error: " + std::to_string(err) + "\n");
     }
 
-
-    if (mode==Read && (dims_from_file[0]=="time" && var_dimensions[0]!="time")) {
-      // For Read operations, we may not consider "time" as a field dimension, so if the
-      // input file has "time", simply disregard it in this check.
-      dims_from_file.erase(dims_from_file.begin());
+    // Here, let's only try to access var_dimensions[0] when we know for sure
+    // that var_dimensions is actually dimensioned (i.e., .size()>0)
+    if (var_dimensions.size()>0) {
```

Comment:
Up to you, but you could also do this:
```
if (mode==Read and dims_from_file[0]=="time" and ekat::contains(var_dimensions,"time")) {
 ...
}
```
I wrote the `ekat::contains(x,y)` function as a shortcut to the usual
```
std::find(x.begin(),x.end(),y)!=x.end()
```
If the input container is empty, it simply returns false.